client/pages/post-form.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import SideSelectButton from '../components/side-select-button';
+
+export default class PostForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      side: 'White',
+      message: ''
+    };
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleMessage = this.handleMessage.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSelect(event) {
+    if (event.target.classList.contains('white')) {
+      this.setState({ side: 'White' });
+    } else if (event.target.classList.contains('black')) {
+      this.setState({ side: 'Black' });
+    }
+  }
+
+  handleMessage(event) {
+    this.setState({ message: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const button = event.nativeEvent.submitter;
+    if (button.classList.contains('side-select')) {
+      return;
+    }
+    if (button.classList.contains('cancel-btn')) {
+      window.location.hash = '#join';
+      return;
+    }
```

Comment:
Try restructuring your React elements so you don't need these guards. You can add `type="button"` to a button element to prevent it from triggering form submission. 

Also, you should use an anchor styled like a button instead of a button if possible.